cloud/services/subnets/subnets.go

```diff
@@ -49,15 +49,15 @@ func (s *Service) getExisting(ctx context.Context, rgName string, spec *Spec) (*
 	var sg infrav1.SecurityGroup
 	if subnet.SubnetPropertiesFormat != nil && subnet.SubnetPropertiesFormat.NetworkSecurityGroup != nil {
 		sg = infrav1.SecurityGroup{
-			Name: to.String(subnet.SubnetPropertiesFormat.NetworkSecurityGroup.Name),
+			Name: spec.SecurityGroupName,
 			ID:   to.String(subnet.SubnetPropertiesFormat.NetworkSecurityGroup.ID),
 			Tags: converters.MapToTags(subnet.SubnetPropertiesFormat.NetworkSecurityGroup.Tags),
```

Comment:
tags should be empty too no?

Reply:
Indeed. I would need to propagate them on the spec. Does that seem reasonable? I'm not sure they are actually settable.

Reply:
they are but in the security group service, not in here... I'm thinking we shouldn't be touching the security group spec in the subnet service but rather do that in the security group service, no? Maybe just put the ID but keep everything else the same. WDYT?

Reply:
I've fixed this, so the full structures of NSG and RT are returned from the GET request via `$expand`. Unfortunately, tags is not returned on that call, so I added a direct call to get the NSG and the RT to fill them with all expected data.